src/dotnet/commands/dotnet-run/LaunchSettings/LaunchSettingsManager.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Tools.Run.LaunchSettings
+{
+    internal class LaunchSettingsManager
+    {
+        private const string ProfilesKey = "profiles";
+        private const string CommandNameKey = "commandName";
+        private const string DefaultProfileCommandName = "Project";
+        private static IReadOnlyDictionary<string, ILaunchSettingsProvider> _providers;
+
+        static LaunchSettingsManager()
+        {
+            _providers = new Dictionary<string, ILaunchSettingsProvider>
+            {
+                { ProjectLaunchSettingsProvider.CommandNameValue, new ProjectLaunchSettingsProvider() }
+            };
+        }
+
+        public static LaunchSettingsApplyResult TryApplyLaunchSettings(string launchSettingsJsonContents, ref ICommand command, string profileName = null)
```

Comment:
Yep, not all the formats are the same for the inner profiles or the top level of configuration parameters. We can set up a skeleton model, but it'd pretty much only contain a "profiles" property with a `Dictionary<string, JObject>` or `Dictionary<string, Dictionary<string, JToken>>` where the actual profile handling would need to re-deserialized portions of the document as the proper model types.